Scripts/auth.js

```diff
@@ -0,0 +1,96 @@
+// auth.js
+import { API_REGISTER_URL, API_LOGIN_URL } from "./common.js";
+
+/**
+ * Registers a new user
+ * @param {string} name User's name
+ * @param {string} email User's email
+ * @param {string} password User's password
+ * @returns {Promise<Object>} The response data
+ */
+export async function register(name, email, password) {
+  if (!name.match(/^\w+$/)) {
+    throw new Error(
+      "Username must not contain punctuation symbols apart from underscore (_)."
+    );
+  }
+
+  if (!email.match(/@(noroff\.no|stud\.noroff\.no)$/)) {
+    throw new Error(
+      "Email must be a valid 'stud.noroff.no' or 'noroff.no' address."
+    );
+  }
+
+  if (password.length < 8) {
+    throw new Error("Password must be at least 8 characters long.");
+  }
+
+  const requestBody = JSON.stringify({ name, email, password });
+  console.log("Register request payload:", requestBody); // Payload
+
+  try {
+    const response = await fetch(API_REGISTER_URL, {
```

Comment:
You are using a try/catch in the function that calls this, so this try/catch here is not necessary.